mitodl/ol-django
# Release management

This PR:

- Updates the README
- Adds a github action workflow for testing/releasing to pypi (you should see one for each python version, plus codedov and a skipped release job at the bottom of this PR)
  - The release job can't really be tested within a branch, so this is a best-attempt at what _should_ work
- Adds an `invoke` command that versions an app and its changelog

To test:

- Install the deps in the README
- **Important:** Comment out the line `pre=[call(checkout_branch, "main")],` in `tasks.py`
Test out the versioning command (install deps as instructed in the README):

`invoke version mitol-django-mail minor`

You should see:
- A new commit, which bumps the project version and versions the items in the changelog under "[Unreleased]"
- That commit with a namespaced version tag

**Important:** clean up your state by doing a hard reset on the tagged commit,  deleting the tag, and uncommenting the line commented out above
